lib/APPSensorFusion/DrivingState.cpp

```diff
@@ -67,6 +67,10 @@
 
 void DrivingState::entry()
 {
+    /* Clear the Odometry Position in favor of reproducibility. */
+    Odometry::getInstance().clearPosition();

```

Comment:
This means the robot always think it starts at (0,0)?
If in reality this is not the case, you would have to transform the recorded data to match the ground truth. Did you take this into account?